service/resource/cloudformation/adaptor.go

```diff
@@ -0,0 +1,85 @@
+package cloudformation
+
+import (
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/microerror"
+	"golang.org/x/sync/errgroup"
+
+	awsutil "github.com/giantswarm/aws-operator/client/aws"
+)
+
+type hydrater func(awstpr.CustomObject, awsutil.Clients) error
+
+type adaptor struct {
+	ASGType string
+
+	// worker launch configuration
+	ImageID                  string
+	SecurityGroupID          string
+	InstanceType             string
+	IAMInstanceProfileName   string
+	BlockDeviceMappings      []BlockDeviceMapping
+	AssociatePublicIPAddress bool
+	SmallCloudConfig         string
+
+	// worker autoscaling group
+	SubnetID               string
+	AZ                     string
+	ASGMinSize             int
+	ASGMaxSize             int
+	LoadBalancerName       string
+	HealthCheckGracePeriod string
+	MinInstancesInService  int
+	MaxBatchSize           string
+	RollingUpdatePauseTime string
+}
+
+type BlockDeviceMapping struct {
+	DeviceName string
+	VolumeSize string
+	VolumeType string
+}
+
+func (a *adaptor) getMain(customObject awstpr.CustomObject, clients awsutil.Clients) error {
+	a.ASGType = "worker"
+
+	var g errgroup.Group
+
+	hydraters := []hydrater{
+		a.getAutoScalingGroup,
+		a.getLaunchConfiguration,
+	}
+
+	// each hydrater works on different data and may need to query AWS API, so that
+	// they are run in separate goroutines
```

Comment:
For a start we'll need to query the AWS API to get data like the subnet IDs. But as we move more resources to CF we should be able to use references.

Having separate goroutines is good but if it's for the API calls then I think that's a temporary state. Lastly very nitpicky but can we make the comments full sentenences.

Reply:
Agree, let's remove the `errGroup` for now until it is actually needed.